pgmpy/models/ClusterGraph.py

```diff
@@ -256,24 +279,16 @@ def check_model(self):
             True if all the checks are passed
         """
         for clique in self.nodes():
-            if self.get_factors(clique):
-                pass
-            else:
-                raise ValueError('Factors for all the cliques or clusters not'
+            factors = filter(lambda x: set(x.scope()) == set(clique), self.factors)
+            if not any(factors):
+                raise ValueError('Factors for all the cliques or clusters not '
                                  'defined.')
 
-        if len(self.factors) != len(self.nodes()):
-            raise ValueError('One to one mapping of factor to clique or cluster'
-                             'is not there.')
-
+        cardinalities = self.get_cardinality()
         for factor in self.factors:
             for variable, cardinality in zip(factor.scope(), factor.cardinality):
-                if ((self.cardinalities[variable]) and
-                        (self.cardinalities[variable] != cardinality)):
+                if (cardinalities[variable] != cardinality):
                     raise CardinalityError(
```

Comment:
@yashu-seth Because most of the cases can be handled by the in built Exceptions (along with the error message). So, we don't need these custom exceptions.